js/tidex2.js

```diff
@@ -0,0 +1,960 @@
+            const currency = response[i];
+            const id = this.safeString (currency, 'symbol');
+            const precision = this.safeInteger (currency, 'amountPoint');
+            const code = this.safeCurrencyCode (id);
+            const active = this.safeValue (currency, 'visible');
+            const withdrawEnable = this.safeValue (currency, 'withdrawEnable');
+            const depositEnable = this.safeValue (currency, 'depositEnable');
+            const name = this.safeString (currency, 'name');
+            const fee = this.safeNumber (currency, 'withdrawFee');
+            result[code] = {
+                'id': id,
+                'code': code,
+                'name': name,
+                'active': active,
+                'deposit': depositEnable,
+                'withdraw': withdrawEnable,
+                'precision': precision,
+                'limits': {
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'withdraw': {
+                        'min': this.safeNumber (currency, 'withdrawMinAmount'),
+                        'max': undefined,
+                    },
+                    'deposit': {
+                        'min': this.safeNumber (currency, 'depositMinAmount'),
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+                'fee': fee,
+                'info': currency,
+            };
+        }
+        return result;
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.publicGetMarkets (params);
+        const markets = this.safeValue (response, 'result');
+        //
+        //     {
+        //         "code":200,
+        //         "success":true,
+        //         "message":"",
+        //         "result":[
+        //             {
+        //                 "name":"BCH_BTC",
+        //                 "moneyPrec":8,
+        //                 "stock":"BCH",
+        //                 "money":"BTC",
+        //                 "stockPrec":8,
+        //                 "feePrec":8,
+        //                 "minAmount":"0.001"
+        //             }
+        //         ]
+        //     }
+        //
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = this.safeString (market, 'name');
+            const [ baseId, quoteId ] = id.split ('_');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const precision = {
+                'amount': undefined,
```

Comment:
I think that by inference we could also populate the `amount` precision field using the `stockPrec` field in the response:

```JavaScript
const precision = {
    'amount': this.safeInteger (market, 'stockPrec'),
    'price': this.safeInteger (market, 'moneyPrec')    
}
```